client/src/components/AlumniDirectory.js

```diff
@@ -228,6 +245,10 @@ export default class AlumniDirectory extends Component {
         this.setState({alumniDetails: this.state.entries[e.currentTarget.dataset.id]})
         this.toggleRequestModal()
     }
+    handleConnectButton(e) {
+        this.setState({alumniDetails: this.state.entries[e.currentTarget.dataset.id]})
```

Comment:
the `this.toggleAlumniContactModal()` might need to happen `this.setState({}, () => { <here> })`